client/core/core.go

```diff
@@ -7362,6 +7375,13 @@ func (c *Core) handleReconnect(host string) {
 		return
 	}
 
+	if atomic.LoadUint32(&dc.anomaliesCount) > wsMaxAnomalyCount {
+		// Send notification to check connectivity.
+		subject, details := c.formatDetails(TopicDexConnectivity, host)
+		c.notify(newConnEventNote(TopicDexConnectivity, subject, host, dc.status(), details, db.Poke))
+		atomic.StoreUint32(&dc.anomaliesCount, 0) // reset anomalies count.
+	}
+
```

Comment:
I'm wondering if we should do this on the **disconnect** instead of the connect. Down where you're doing 
```Go
atomic.AddUint32(&dc.anomaliesCount, 1)
```

Reply:
That would also be nice because with mods done in a single place, it could be properly atomic with `if atomic.CompareAndSwapUint32(&dc.anomaliesCount, wsMaxAnomalyCount, 0) { /* notify */ }`

Reply:
Although now that I say that, it could just be a `atomic.AddUin32(&dc.anomaliesCount, 1) % wsMaxAnomalyCount == 0`